WoW-Realm-Tracker/ViewControllers/FavoritesViewController.swift

```diff
@@ -83,8 +84,8 @@ class FavoritesViewController: UITableViewController {
 }
 
 extension FavoritesViewController: RealmsDelegate {
-    func receivedRealms(realms: [Realm]) {
-        controller = FavoriteRealmsController(realms: realms)
+    func receivedRealms(response: Result<[Realm], NSError>) {
+        response.map { realms in self.controller = FavoriteRealmsController(realms: realms) }
```

Comment:
fwiw, I'd _not_ use `map` for side effects (ignoring the return value), but that's an ongoing argument with no good resolution.

I _would_ be in favor of doing something like this:

```swift
let controller = response.map(FavoriteRealmsController.init) // you can pass initializers as static methods in Swift 2.0
if let controller = controller {
  self.controller = controller
}
```

shadowing is a bit weird there, but you get the gist.